explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -53,119 +54,83 @@
+    clientExceptionContractNotFound = new ClientException("contract not found", StatusCode.CONTRACT_NOT_FOUND);
+
     explorer = new Explorer(clientService, clientConfig);
     when(clientConfig.getCertHolderId()).thenReturn(HOLDER_ID);
-
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseOK);
-    when(clientService.registerContract(
-            anyString(), anyString(), anyString(), eq(Optional.empty())))
-        .thenReturn(ledgerServiceResponseOK);
-
-    when(ledgerServiceResponseOK.getStatus()).thenReturn(StatusCode.OK.get());
-    when(ledgerServiceResponseCertificateNotFound.getStatus())
-        .thenReturn(StatusCode.CERTIFICATE_NOT_FOUND.get());
-    when(ledgerServiceResponseCertificateNotFound.getMessage()).thenReturn("certificate not found");
-
-    when(contractExecutionResponseOK.getStatus()).thenReturn(StatusCode.OK.get());
-    when(contractExecutionResponseCertificateNotFound.getStatus())
-        .thenReturn(StatusCode.CERTIFICATE_NOT_FOUND.get());
-    when(contractExecutionResponseCertificateNotFound.getMessage())
-        .thenReturn("certificate not found");
-    when(contractExecutionResponseContractNotFound.getStatus())
-        .thenReturn(StatusCode.CONTRACT_NOT_FOUND.get());
-    when(contractExecutionResponseContractNotFound.getMessage()).thenReturn("contract not found");
   }
 
   @Test
   public void get_CertificateAndContractNotRegistered_ShouldSucceed() {
     // Arrange
-    when(contractExecutionResponseOK.getResult()).thenReturn("{ \"id\": \"bar\" }");
+    Optional<JsonObject> jsonObject = Optional.of(Json.createObjectBuilder().add("status", StatusCode.OK.get()).build());
+    when(contractExecutionResultOK.getResult()).thenReturn(jsonObject);
     when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseContractNotFound,
-            contractExecutionResponseOK);
-
+        .thenReturn(contractExecutionResultOK);
     // Act
     JsonObject result = explorer.get(ASSET_ID);
 
     // Assert
-    verify(clientService, Mockito.times(3))
+    verify(clientService)
         .executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class));
-    assertThat(result.getString("id")).isEqualTo("bar");
+    assertThat(result).isEqualTo(jsonObject.get());
   }
 
   @Test
   public void get_FailedToRegisterCertificate_ShouldThrowExplorerException() {
     // Arrange
-    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
-        .thenReturn(contractExecutionResponseCertificateNotFound);
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseCertificateNotFound);
+    when(clientService.executeContract(eq( HOLDER_ID + "GET"), any(JsonObject.class)))
+        .thenThrow(clientExceptionCertificateNotFound);
+    doAnswer((i) -> { return null;} ).when(clientService).registerCertificate();
```

Comment:
Why is it returning null ?
Should it be like this ?
```suggestion
    doNothing().when(clientService).registerCertificate();
```

Reply:
Thank you for review comment.
Code updated with the above modification.
It's my mistake in between quick solve.